pysat/tests/classes/cls_instrument_property.py

```diff
@@ -0,0 +1,648 @@
+        return
+
+    def test_str_w_orbit(self):
+        """Test string output with Orbit data."""
+
+        reload(pysat.instruments.pysat_testing)
+        orbit_info = {'index': 'mlt',
+                      'kind': 'local time',
+                      'period': np.timedelta64(97, 'm')}
+        testInst = pysat.Instrument(platform='pysat', name='testing',
+                                    num_samples=10,
+                                    clean_level='clean',
+                                    update_files=True,
+                                    orbit_info=orbit_info)
+
+        self.out = testInst.__str__()
+
+        # Check that orbit info is passed through
+        assert self.out.find('Orbit Settings') > 0
+        assert self.out.find(orbit_info['kind']) > 0
+        assert self.out.find('Loaded Orbit Number: 0') > 0
+
+        # Activate orbits, check that message has changed
+        testInst.load(self.ref_time.year, self.ref_doy)
+        testInst.orbits.next()
+        self.out = testInst.__str__()
+        assert self.out.find('Loaded Orbit Number: 1') > 0
+        return
+
+    def test_str_w_padding(self):
+        """Test string output with data padding."""
+
+        self.testInst.pad = dt.timedelta(minutes=5)
+        self.out = self.testInst.__str__()
+        assert self.out.find('Data Padding: 0:05:00') > 0
+        return
+
+    def test_str_w_custom_func(self):
+        """Test string output with custom function."""
+
+        def passfunc(self):
+            pass
+        self.testInst.custom_attach(passfunc)
+        self.out = self.testInst.__str__()
+        assert self.out.find('passfunc') > 0
+        return
+
+    def test_str_w_load_lots_data(self):
+        """Test string output with loaded data."""
+
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.out = self.testInst.__str__()
+        assert self.out.find('Number of variables:') > 0
+        assert self.out.find('...') > 0
+        return
+
+    def test_str_w_load_less_data(self):
+        """Test string output with loaded data."""
+
+        # Load the test data
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+
+        # Ensure the desired data variable is present and delete all others
+        # 4-6 variables are needed to test all lines; choose the lesser limit
+        nvar = 4
+        self.testInst.data = self.testInst.data[self.testInst.variables[:nvar]]
+
+        # Test output with one data variable
+        self.out = self.testInst.__str__()
+        assert self.out.find('Number of variables: 4') > 0
+        assert self.out.find('Variable Names') > 0
+        for n in range(nvar):
+            assert self.out.find(self.testInst.variables[n]) > 0
+        return
+
+    def test_instrument_init(self):
+        """Test if init function supplied by instrument can modify object."""
+
+        assert self.testInst.new_thing
+        return
+
+    def test_custom_instrument_load(self):
```

Comment:
Do we still need this test? There are general load tests a couple files above.

Differences Spotted
 - I do see it does check for loading by day of year outside January, so that is good. 
 - Also has instantiation by inst_module and then a load. I believe this general process is covered by running TestBasics with a inst_module instantiated Instrument.

My current thinking is that we may want to parameterize the basic loading test way above with a variety of dates in a normal year (there is a leap year test for doy 366).

Reply:
we can probably drop this

Reply:
dropped